### PR TITLE
[KYUUBI #1522] Pass the client ip to engine

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

Pass the client ip to engine. For details: #1522

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -78,7 +79,7 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
       protocol,
       user,
       password,
-      ipAddress,
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:81]: Seems the ipAddress of the kyuubi server is also useful if we deploy in ha. We can get the information which Kyuubi server opened the engine session.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/package.scala [hunk 2]
```diff
@@ -27,4 +27,6 @@ package object session {
   val METACONF_PREFIX = "metaconf:"
 
   val SPARK_PREFIX = "spark."
+
+  val CLIENT_IP_KEY = "kyuubi.clent.ip"
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/package.scala:31]: client

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala (hunk 1 above)
Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:81]: > Seems the ipAddress of the kyuubi server is also useful if we deploy in ha. We can get the information which Kyuubi server opened the engine session.

Yes, but the current usage is more inclined to clientip, do we need to add an extra ip field?

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala (hunk 1 above)
Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:81]: We can do this in a new PR, it's better to show client ip at UI side . And just record the kyuubi server ipAddress in event log.

Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:81]: OK, I'll record an issue first.

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**